include/uart.h

```diff
@@ -245,6 +384,157 @@ struct uart_driver_api {
 
 };
 
+#ifdef CONFIG_UART_HIGH_LEVEL_API
+
+/**
+ * @brief Set event handler function.
+ *
+ * @param dev       UART device structure.
+ * @param callback  Event handler.
+ * @param user_data Data to pass to event handler function.
+ *
+ * @retval 0	    If successful, negative errno code otherwise.
+ */
+static inline int uart_callback_set(struct device *dev,
+				    uart_callback_t callback,
+				    void *user_data)
+{
+	const struct uart_driver_api *api =
+			(const struct uart_driver_api *)dev->driver_api;
+
+	return api->callback_set(dev, callback, user_data);
+}
+
+/**
+ * @brief Send given number of bytes from buffer through UART.
+ *
+ * Function returns immediately and event handler,
+ * set using @ref uart_set_callback, is called after transfer is finished.
+ *
+ * @param dev     UART device structure.
+ * @param buf     Pointer to transmit buffer.
+ * @param len     Length of transmit buffer.
+ * @param timeout Timeout in milliseconds.
+ *
+ * @retval enum uart_error In case of error in blocking mode operation.
+ * @retval -EBUSY	   There is already an ongoing transfer.
+ * @retval 0		   If successful, negative errno code otherwise.
+ */
+__syscall int uart_tx(struct device *dev, u8_t *buf, size_t len, u32_t timeout);
```

Comment:
is there a reason for having TX timeout here and not as `tx_timeout_set`?

Reply:
RX timeout and TX timeout work in a different way, TX timeout will be generated if TX has not finished in given time, RX timeout will be generated if after receiving some chars, no more chars have been received for given time, RX timeout can be generated multiple times, while TX timeout will be generated only once per transmission, thats why I think that TX timeout value should be binded with starting TX transfer

Reply:
Removed syscall